src/inmanta/protocol/methods.py

```diff
@@ -480,6 +480,41 @@ def resource_action_update(
 # Manage configuration model versions
 
 
+@typedmethod(
+    path="/discovered_resources",
+    validate_sid=False,
+    operation="POST",
+    agent_server=True,
+    arg_options=ENV_OPTS,
+    client_types=[const.ClientType.agent],
+)
+def discovered_resources_create(env: uuid.UUID, discovered_resource_name: str, value: List[Dict[str, str]]) -> None:
+    """
+    Send discovered resource to the server
+
+    :param env: The id of the environment this resource belongs to
+    :param discovered_resource_name: The resource with the given resource_version_id id from the agent
+    """
+
+
+@typedmethod(
+    path="/discovered_resources",
+    validate_sid=False,
+    operation="GET",
+    agent_server=True,
+    arg_options=ENV_OPTS,
+    client_types=[const.ClientType.agent],
```

Comment:
why agent? I would expect this to be api? I suspect this will allow the removal of the two other options above I flagged.